pkg/key/cluster.go

```diff
@@ -17,9 +17,17 @@ func ClusterConfigMapName(customResource v1alpha1.App) string {
 		return ingressControllerConfigMapName
 	}
 
+	if IsInOrgNamespace(customResource) {
+		return fmt.Sprintf("%s-cluster-values", ClusterLabel(customResource))
```

Comment:
This makes me worried a bit, since there is a risk `ClusterLabel` returns an empty value. On the other hand, the recent changes to both, the app package and admission controller, were introduced to make sure the `cluster` label is present if app has been admitted.

Reply:
Good to check and the mutation webhook is triggered before the validation webhook. So if the cluster label is missing we should catch it.

However like we discussed I think it would be good to have integration coverage in app-admission-controller for app CRs in both cluster and org namespaces.

Reply:
@rossf7 So, let's put integration tests aside for a moment, what would be the expected behaviour of the mutation logic in app admission controller? To detect the missing `giantswarm.io/cluster` label and thus fail early on, and when it's set skip some of the version label-related logic, is that correct?

Reply:
Yes, I don't think we can default the cluster label. There will be multiple clusters in the org so we need the user to specify which one. If they don't then the validation webhook should return an error.

Just to add context for defaulting the version label for the cluster namespace this is possible. We use the namespace to find the chart-operator app CR and use its version since it always exists.

https://github.com/giantswarm/app-admission-controller/blob/master/pkg/app/mutate_app.go#L113-L116